add support for oci based helm repo

Currently, when we use OCI based helm repo like `repoURL: public.ecr.aws/aws-ec2/helm`, we receive the error 

`
Error: could not find protocol handler for: 
Could not process the source Application: failed to download chart
panic: Could not process the source Application: failed to download chart
`

The problem is in how the repo URL is handled here https://github.com/shini4i/argo-compare/blob/main/cmd/argo-compare/utils/helm-chart-processor.go#L98

To make it work the command should include oci: `helm pull oci://my-oci-repo.example.com/my-chart --version 1.0.0`

--- a/cmd/argo-compare/utils/helm-chart-processor.go
+++ b/cmd/argo-compare/utils/helm-chart-processor.go
@@ -92,7 +92,7 @@ func (g RealHelmChartProcessor) DownloadHelmChart(cmdRunner interfaces.CmdRunner
 			cyan(chartName))
 
 		// we assume that if repoUrl does not have protocol, it is an OCI helm registry
-		// hence we mutate the content on chartName and remove content of repoUrl
+		// hence we mutate the content of chartName and remove content of repoUrl
 		if !strings.Contains(repoUrl, "http") {
 			chartName = fmt.Sprintf("oci://%s/%s", repoUrl, chartName)
 			repoUrl = ""

--- a/cmd/argo-compare/utils/helm-chart-processor.go
+++ b/cmd/argo-compare/utils/helm-chart-processor.go
@@ -4,6 +4,7 @@ import (
 	"errors"
 	"fmt"
 	"os"
+	"strings"
 
 	"github.com/fatih/color"
 	"github.com/shini4i/argo-compare/internal/helpers"
@@ -89,6 +90,13 @@ func (g RealHelmChartProcessor) DownloadHelmChart(cmdRunner interfaces.CmdRunner
 		g.Log.Debugf("Downloading version [%s] of [%s] chart...",
 			cyan(targetRevision),
 			cyan(chartName))
+
+		// we assume that if repoUrl does not have protocol, it is an OCI helm registry
+		// hence we mutate the content on chartName and remove content of repoUrl
+		if !strings.Contains(repoUrl, "http") {
+			chartName = fmt.Sprintf("oci://%s/%s", repoUrl, chartName)
+			repoUrl = ""
+		}
 
 		stdout, stderr, err := cmdRunner.Run("helm",
 			"pull",

